sisl/io/orca/txt.py

```diff
@@ -0,0 +1,207 @@
+            else:
+                return None
+        return self._no
+
+    @sile_fh_open()
+    def read_electrons(self, all=False):
+        """ Read number of electrons (alpha, beta)
+
+        Parameters
+        ----------
+        all: bool, optional
+            return electron numbers from all steps (instead of last)
+        """
+        def readE(itt, reread=True):
+            f = self.step_to("Number of Alpha Electrons", reread=reread)
+            if f[0]:
+                alpha = float(f[1].split()[-1])
+                beta = float(next(itt).split()[-1])
+            else:
+                return None
+            return alpha, beta
+
+        itt = iter(self)
+        E = []
+        e = readE(itt)
+        while e is not None:
+            E.append(e)
+            e = readE(itt, reread=False)
+
+        if all:
+            return np.array(E)
+        if len(E) > 0:
+            return np.array(E[-1])
+        return None
+
+    @sile_fh_open()
+    def read_energy(self, all=False, convert=True):
+        """ Reads the energy specification from ORCA property.txt file.
+
+        Parameters
+        ----------
+        all: bool, optional
+            return a list of dictionaries from each step
+        convert: bool, optional
+            convert from Hartree to eV units
+
+        Returns
+        -------
+        PropertyDict : all data from the "DFT_Energy" segment
+        """
+
+        Hartree2eV = 27.2113834
+
+        def readE(itt, reread=True):
+            # read the DFT_Energy block
+            f = self.step_to("$ DFT_Energy", reread=reread)[0]
+            if not f:
+                return None
+            next(itt) # description
+            next(itt) # geom. index
+            next(itt) # prop. index
+            line = next(itt)
+            E = PropertyDict()
+            while "----" not in line:
+                v = line.split()
+                value = float(v[-1])
+                if convert:
+                    value *= Hartree2eV
+                if v[0] == "Exchange":
+                    E["exchange"] = value
+                elif v[0] == "Correlation":
+                    if v[2] == "NL":
+                        E["correlation_nl"] = value
+                    else:
+                        E["correlation"] = value
+                elif v[0] == "Exchange-Correlation":
+                    E["xc"] = value
+                elif v[0] == "Embedding":
+                    E["embedding"] = value
+                elif v[1] == "DFT":
+                    E["total"] = value
+                line = next(itt)
+            line = next(itt)
+            if "$ VdW_Correction" in line:
+                v = self.step_to("Van der Waals Correction:", reread=reread)[1].split()
+                value = float(v[-1])
+                if convert:
+                    value *= Hartree2eV
+                E["vdw"] = value
+                E["total_vdw"] = E["total"] + value
```

Comment:
I must admit I don't really like this, perhaps just clarify in the documentation that the total energy is without vdw. Then users should do it them selves. Manipulating energies on output I don't really like.

Reply:
I understand, but here I want to stay as close as possible to the values printed by ORCA. To me it appears that the vdW correction is *not* explicitly added to the DFT total energy. I do not know why.

Reply:
Oh, I probably misunderstood. You mean that I should simply avoid this entry?